velox/exec/tests/utils/TestIndexStorageConnector.cpp

```diff
@@ -0,0 +1,258 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/exec/tests/utils/TestIndexStorageConnector.h"
+
+namespace facebook::velox::exec::test {
+
+TestIndexSource::TestIndexSource(
+    const facebook::velox::RowTypePtr& inputType,
+    const facebook::velox::RowTypePtr& outputType,
+    const std::shared_ptr<facebook::velox::connector::ConnectorTableHandle>&
+        tableHandle,
+    facebook::velox::connector::ConnectorQueryCtx* connectorQueryCtx,
+    folly::Executor* executor)
+    : tableHandle_(
+          std::dynamic_pointer_cast<TestIndexTableHandle>(tableHandle)),
+      inputType_(inputType),
+      outputType_(outputType),
+      keyType_(tableHandle_->indexTable()->keyType),
```

Comment:
tableHandle_ can be null, which will cause a crash

perhaps, change the caller to do the cast and check, then pass TestIndexTableHandle to TestIndexSource's ctor.